line_detection_updated.py

```diff
@@ -0,0 +1,302 @@
+    lines : `numpy.ndarray`
+        Polar coordinates of the detected lines
+    bandwidth: 'int', optional, default=50
+        Provides the bandwidth parameter for MeanShift algorithm
+
+    Returns
+    --------
+    clusteredLines : `numpy.ndarray`
+        An array where each row has the polar coordinates of an estimated line
+        and the cluster label for that line
+    """
+    # Convert to an array
+    cart_coords_array = np.array(cart_coords)
+
+    # Apply MeanShift clustering
+    clustering = MeanShift(bandwidth=bandwidth).fit(cart_coords_array)
+    labels = (clustering.labels_).reshape((len(clustering.labels_), 1))
+    clustered_lines = np.hstack((lines, labels))
+
+    # Plot the clusters
+    for lab in set(clustering.labels_):
+        mask = clustering.labels_ == lab
+        x = cart_coords_array[:, 0]
+        y = cart_coords_array[:, 1]
+
+        plt.scatter(x[mask], y[mask], label=lab)
+        plt.xlabel("x coordinate")
+        plt.ylabel("y coordinate")
+        plt.legend()
+
+    return clustered_lines
+
+
+# Create class
+class LineDetection:
+    '''
+    Applies noise reduction image processing to the raw image, finds edges using Canny and
+    detects straight lines using Scikit Image hough_line and hough_line_peaks
+    '''
+
+    # Instatiating constructors
+    def __init__(self):
+        '''
+        Initializes the class' parameters
+
+        Parameters
+        ----------
+        image : 'numpy.ndarray'
+            The raw image
+        mask : `bool`, default=False
+            Indicates whether or not to mask the image
+        mask_percent: 'float', default=0.2
+            The proportion of the image's edge to be masked
+        nstd1_cut: 'float', default=2
+            Applied on the raw image. Pixel intensity above these many standard deviations are
+            reduced to the intensity value at that standard deviation. Pixels below these many
+            standard deviations are converted to 0
+        nstd2_binary_cut: 'float', default=0.5
+            Applied on the processed image. The pixel intensity at these many standard
+            deviations above the mean becomes the threshold limit. A binary threshold of
+            0/255 is applied at the limit value
+        threshold: 'float', default=0.075
+            The voting threshold for Hough Line Peaks as a proportion of the length of
+            the diagonal
+        flux_prop_thresholds: 'list', default=[0.1,0.2,0.3,1]
+            Lists the possible thresholds for proportion of bright pixels in the image,
+            where a pixel is considered bright if it's intensity is above the mean
+        blur_kernel_sizes: 'list', default=[3,5,9,11]
+            Decides the kernel size for blurring the thresholded image and corresponds
+            element wise to the flux_prop_thresholds list. Must be of the same length
+            as flux_prop_thresholds
+        '''
+        # Assign the raw image
```

Comment:
The general suggestion is that the `__init__` (and other private functions and special overloads) are documented in the class docstring. So if you move this up to where you document class and just get rid of the "initialized class' parameters line" you should be fine. See [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html#documenting-classes) style guide.

Also you document all these parameters to the `__init__` function, but init itself doesn't take any parameters in. Either make the init function take these parameters in, and sets their default values, or rename this whole paragraph to `Attributes`, since at the moment they are not parameters but class attributes.
I would prefer that they become parameters.

On a different broader note, some of these parameter names are so obscure they really don't reveal much about themselves at all. Instead of `nstd_cut` we use something like `brightness_cuts` and then int he documentation state.
```
Limits of pixel brightness values that will be cut. 
Pixels with brightness value larger than `mean - limit * std` 
are set to 0, and pixels with `mean + limit*std` 
are set to `mean + limit*std`.
```
Bonus point if we let this be an `tuple` of the bottom and top limits too. This just sound much less confusing in my head. Same goes for the `nstd2_binary_cut` - I'm just not sure what this parameter is doing from that documentation.

Reply:
Kept the initiators as attributes for now, can change later. 
Edited names of the brightness cuts, converted the first one to a tuple. The other one is just a binary threshold, so just has one value.